python/paddle/fluid/tests/unittests/test_while_loop_op.py

```diff
@@ -110,24 +113,146 @@ def internal_body(j, init, sums):
             loop_len2 = layers.fill_constant(shape=[1], dtype='int64', value=3)
             ones = layers.fill_constant(shape=[3, 3], dtype='float32', value=1)
 
-            res = layers.while_loop(external_cond, external_body,
+            out = layers.while_loop(external_cond, external_body,
                                     [i, j, init, sums])
 
             data = np.random.rand(3, 3).astype('float32')
             data_sums = np.zeros([3, 3]).astype('float32')
 
-        place = fluid.CPUPlace()
+        place = fluid.CUDAPlace(0) if core.is_compiled_with_cuda(
+        ) else fluid.CPUPlace()
         exe = fluid.Executor(place)
-        ret = exe.run(main_program,
+        res = exe.run(main_program,
                       feed={'init': data,
                             'sums': data_sums},
-                      fetch_list=res)
+                      fetch_list=out)
         for i in range(3):
             data = np.add(data, 1)
             data_sums = np.add(data, data_sums)
         for j in range(2):
             data_sums = np.add(data, data_sums)
-        self.assertTrue(np.allclose(np.asarray(ret[3]), data_sums))
+        self.assertTrue(np.allclose(np.asarray(res[3]), data_sums))
+
+
+class TestApiWhileLoop_Backward(unittest.TestCase):
+    def test_while_loop_backward(self):
+        def cond(i, x):
+            return layers.less_than(i, ten)
+
+        def body(i, x):
+            x = layers.elementwise_add(x=x, y=one)
+            i = layers.increment(i)
+            return [i, x]
+
+        program = fluid.Program()
+        with fluid.program_guard(program):
+            i = layers.fill_constant(shape=[1], dtype='int64', value=0)
+            ten = layers.fill_constant(shape=[1], dtype='int64', value=10)
+            one = layers.fill_constant(shape=[1], dtype='float32', value=1)
+            x = layers.fill_constant(shape=[1], dtype='float32', value=0)
+            x.stop_gradient = False
+
+            out = layers.while_loop(cond, body, [i, x])
+            mean = layers.mean(out[1])
+            append_backward(mean)
+
+        place = fluid.CUDAPlace(0) if core.is_compiled_with_cuda(
+        ) else fluid.CPUPlace()
+        exe = fluid.Executor(place)
+
+        feed_x = np.asarray([10]).astype('float32')
+        res = exe.run(program, feed={}, fetch_list=[mean.name, mean.grad_name])
```

Comment:
I saw you fetched mean and mean.grad_name but compare it with feed_x and x_grad, is it an error? If yes, please change them, compare the grad value of x or mean value.

Reply:
Done.